common/unsafe/src/main/java/org/apache/spark/unsafe/Platform.java

```diff
@@ -244,6 +251,18 @@ public static void throwException(Throwable t) {
       LONG_ARRAY_OFFSET = _UNSAFE.arrayBaseOffset(long[].class);
       FLOAT_ARRAY_OFFSET = _UNSAFE.arrayBaseOffset(float[].class);
       DOUBLE_ARRAY_OFFSET = _UNSAFE.arrayBaseOffset(double[].class);
+
+      // determine whether double access should be aligned.
+      String arch = System.getProperty("os.arch", "");
+      if (arch.matches("^(arm|arm32)")) {
```

Comment:
@kiszk I have tested on ARM 64 (`aarch64`). [Any alignment works for double access there](http://infocenter.arm.com/help/index.jsp?topic=/com.arm.doc.den0024a/ch08s02.html), though 8-byte aligned access looks to be about 10% faster than unaligned access. Using an intermediate long buffer (your idea) is about 5% slower than direct access regardless of being aligned or not. In both cases, I tested with an ODROID C2 using Oracle Java 8.